test/supervisor_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule KafkaEx.Supervisor.Test do
+  use ExUnit.Case, async: false
+
+  test "uses configured max_restarts" do
+    # stop the application, configure supervision max_restarts and restart the application
+    configured_max_restarts = 15
+
+    :ok = Application.stop(:kafka_ex)
+    Application.put_env(:kafka_ex, :max_restarts, configured_max_restarts)
+    {:ok, _} = Application.ensure_all_started(:kafka_ex)
+    [max_restarts, _max_seconds] = :sys.get_state(KafkaEx.Supervisor) |> elem(9)
```

Comment:
Heh.  This is failing for me with Erlang R18.3 and Elixir 1.2.3.  This is the state I get back:

```
{:state, {:local, KafkaEx.Supervisor}, :simple_one_for_one, [{:child, :undefined, KafkaEx.Server, {KafkaEx.Server, :start_link, []}, :permanent, 5000, :worker, [KafkaEx.Server]}], {:dict, 1, 16, 16, 8, 80, 48, {[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []}, {{[], [[#PID<0.264.0>, [uris: [{"localhost", 9092}, {"localhost", 9093}, {"localhost", 9094}], consumer_group: "kafka_ex"], KafkaEx.Server]], [], [], [], [], [], [], [], [], [], [], [], [], [], []}}}, 15, 120, [], 0, KafkaEx.Supervisor, [15, 120]}
```